AElf.Kernel.Core/Blockchain/Application/IBlockchainService.cs

```diff
@@ -248,6 +248,45 @@ public async Task SetBestChainAsync(Chain chain, long bestChainHeight, Hash best
                 BlockHeight = irreversibleBlockHeight
             };
             await _chainManager.SetIrreversibleBlockAsync(chain, irreversibleBlockHash);
+
+            // Clean branches and not linked
+            var bestChainKey = chain.BestChainHash.ToStorageKey();
```

Comment:
move these codes into chainManager

1. chainManager.CleanBranches(Hash libBlockHash, long libHeight) returns List<Hash>
2. remove blocks, transaction in blockchain service.

Reply:
1. In this logic, need to use method GetBlockHashByHeightAsync, can I move the method GetBlockHashByHeightAsync to manager too?
2. I think we can't remove transaction, because one transaction maybe in two blocks in different branch.